server/middleware/counselling-service.middleware.ts

```diff
@@ -34,6 +34,36 @@ const isValidDelivery: CustomValidator = delivery => {
     }
 }
 
+const isValidHour: CustomValidator = hours => {
+    const days = new Set(['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun'])
+
+    if (Object.keys(hours).length < 7 || Object.keys(hours).length > 7) {
+        return false;
+    }
+
+    for (const day in hours) {
+        if (!(days.has(day))) {
+            return false;
+        } else if (hours[day].length === 1) {
+            if (hours[day][0] !== 0 || hours[day][0] !== 1) {
+                return false;
```

Comment:
@ImaadJ10 When I use the following input:
```
"hours": {
        "mon": [1],
        "tue": [1],
        "wed": [1],
        "thu": [1],
        "fri": [1],
        "sat": [1],
        "sun": [1]
    }
```

The request fails, and I found that the validation function returns false here. Should be a simple fix, but I would double check some other cases too.

Reply:
I'll take a look at it first thing tmrw morning!

Reply:
Just looking at the code now, in my else condition I accidentally am returning 'false' instead of 'true', I'll fix that and add some other tests to make sure it's working